### ISSUE TITLE
[DECOHack] CLEAR STATES in an Each Thing Clause does not Clear States

### ISSUE THREAD

**Issue by MTrop** [MTrop/DoomTools#115]
What the title says.

It's probably "clearing the states," but on a Thing Template, it probably needs to explicitly write that those states are 0 so that it has stuff to do.

Sample Code (from **Aurelius**):
```
each thing from MTF_SSNAZI to MTF_SPAWNFIRE
{
  clear properties
  clear flags
  clear sounds
  clear states
}
```

**Comment by MTrop** [comment]
From `DEHThingTemplate.java`:
```
@Override
public DEHThingTemplate clearLabels()
{
	stateIndexMap.clear();
	// If a template clears labels, explicitly set state 0.
	setSpawnFrameIndex(0);
	setWalkFrameIndex(0);
	setPainFrameIndex(0);
	setMeleeFrameIndex(0);
	setMissileFrameIndex(0);
	setDeathFrameIndex(0);
	setExtremeDeathFrameIndex(0);
	setRaiseFrameIndex(0);
	return this;
}
```

Hmmm. Is there code somewhere that isn't setting these states properly?

**Comment by MTrop** [comment]
Ah. Here it is, in `setLabel()`:
```
@Override
public DEHThingTemplate setLabel(String label, int index)
{
	if (index < 0)
		throw new IllegalArgumentException("index cannot be < 0");
	
	if (index == 0)
		stateIndexMap.remove(label);
	else
		stateIndexMap.put(label, index);
	return this;
}
```

The template should be setting those values explicitly in the table for application later!

**Comment by MTrop** [comment]
Fixed in commit 18bfc5a002bdda5254311284c812f2a407b54084.
